app/models/option_set.rb

```diff
@@ -293,6 +293,23 @@ def as_json(options = {})
     end
   end
 
+  def worksheet_name
+    name = self.name
+    name = name.truncate(31) if self.name.size > 31
+    name = name.gsub(
+      /[\[\]\*\\?\:\/]/, {
+        '[' => '(',
+        ']' => ')',
+        '*' => '∗',
+        '?' => '',
+        ':' => '-',
+        '\\' => '-',
+        '/' => '-'
+      }
+    )
+    name
+  end
+
```

Comment:
I mean you could always do a PORO -- WorksheetBuilder or something. But like I say, not a big deal.